lib/dbservice/users.ex

```diff
@@ -177,6 +177,38 @@ defmodule Dbservice.Users do
     Repo.get_by(Student, student_id: student_id)
   end
 
+  @doc """
+  Gets a student by either student_id or apaar_id.
+  Returns the first student found with either identifier.
+
+  ## Examples
+
+      iex> get_student_by_id_or_apaar_id(%{"student_id" => "1234", "apaar_id" => nil})
+      %Student{}
+      iex> get_student_by_id_or_apaar_id(%{"student_id" => nil, "apaar_id" => "123456789101"})
+      %Student{}
+      iex> get_student_by_id_or_apaar_id(%{"student_id" => nil, "apaar_id" => nil})
+      nil
+  """
+  def get_student_by_id_or_apaar_id(%{"student_id" => student_id, "apaar_id" => apaar_id}) do
+    cond do
+      student_id && student_id != "" ->
+        get_student_by_student_id(student_id)
```

Comment:
Why are the two ways of getting student by `student_id` or `apaar_id` different? One is a method, and one uses `Repo.get`?

Reply:
The `get_student_by_student_id method` also uses `Repo.get`, and since it’s used in multiple places, I just reused it here for consistency. We can refactor it later to handle everything through a single function.